src/sedcli_main.c

```diff
@@ -404,6 +424,20 @@ int write_mbr_handle_opts(char *opt, char **arg)
 	return 0;
 }
 
+int blocksid_handle_opts(char *opt, char **arg)
+{
+	int hwreset_flag = 0; /* No reset BlockSID upon power events */
+
+	if (!strcmp(opt, "device")) {
+		strncpy(opts->dev_path, arg[0], PATH_MAX - 1);
+	} else if (!strcmp(opt, "hwreset")) {
+		hwreset_flag = atoi(arg[0]);
```

Comment:
atoi will accept some random text e.g. "./sedcli -B -d /dev/nvme0n1 -r adsfdadaf" -- it needs to be changed

Reply:
You can use strtoul for a safer conversion.

Reply:
if I use CLI_OPTION_RANGE_INT with min and max, I don't believe I need to change here anymore, please confirm

Reply:
random string is still passed to atoi() and accepted by app. I see 2 possibilities to fix it:
1. add appropriate checks in blocksid_handle_opts() and rejected invalid input
2. more general fix in args parse to prevent invalid inputs for INT
while 2 is more general and prefered approach; 1 is good tradeoff to introduce this feature quickly. I recommend 1

Reply:
Please check my updated code